dask/dataframe/io/parquet.py

```diff
@@ -74,6 +74,11 @@ def _read_fastparquet(fs, paths, myopen, columns=None, filters=None,
     elif len(minmax) > 1:
         if index:
             index_col = index
+        elif index is None:
+            # attempt auto-index
+            index = pf._get_index()
+            if index:
+                index_col = index
         elif 'index' in minmax:
             index_col = 'index'
```

Comment:
> In addition, this might happen to be the right thing to do in many cases, where the data has come from some other table-based data system.

This seems like a logical leap to me that needs more justification. The problem is that the pandas index _is not data_, so if you have a column named "index" that happens to be sorted, that column will appear to "disappear" from the resulting dask.dataframe in the default behavior here (relating to what I showed in #2222). So `df['index']` will fail.

I think the index inference logic is useful and I agree that the performance benefits are worthwhile, but I think having it on by default is a recipe for bugs or rough edges. I think it would be better for the user to explicitly opt in to this behavior. It would be good to solicit some other opinions about this topic